src/ui/public/visualize/visualize.js

```diff
@@ -144,7 +144,12 @@ uiModules
         }
 
         if (oldVis) $scope.renderbot = null;
-        if (vis) $scope.renderbot = vis.type.createRenderbot(vis, $visEl, $scope.uiState);
+        if (vis) {
+          vis.on('renderComplete', () => {
+            $scope.$emit('renderComplete');
+          });
```

Comment:
@w33ble are you ok with removing emiting renderComplete on scope (so you would need to subscribe to $visualizeScope.vis.on instead of $visualizeScope.on  ?